p2panda-rs/src/test_utils/db/validation.rs

```diff
@@ -0,0 +1,432 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+//! Methods for validating entries and operations against expected and stored values.
```

Comment:
So all of this is here is tooling for the massive storage tests? :rofl: Its amazing stuff! I'm curious to see how we can integrate it into the library at one point.

Reply:
Well, it's the module we use in `aquadoggo`, both `domain` and `validation` are directly picked up from there. I brought them across as then I could recycle the existing code for the test utils and we'd be fairly confident the behaviour of `MemoryStore` and `SqlStore` are equal.

But in the context of `p2panda-rs`, yes, is all just test tooling :rofl: I had imagined they would come into the main lib eventually, but thought at this point it wasn't necessary and just having them here was enough.

Reply:
(i wouldn't have written all this just for test utils.... honest)